google/resource_bigquery_dataset_test.go

```diff
@@ -112,3 +159,110 @@ resource "google_bigquery_dataset" "test" {
   }
 }`, datasetID)
 }
+
+func testAccBigQueryDatasetWithOneAccess(datasetID string) string {
+	return fmt.Sprintf(`
+resource "google_bigquery_dataset" "access_test" {
+  dataset_id                  = "%s"
+  friendly_name               = "foo"
+  description                 = "This is a foo description"
+  location                    = "EU"
+  default_table_expiration_ms = 3600000
+
+  access = [
+    {
+	role          = "OWNER"
+	user_by_email = "Joe@example.com"
+    }
+  ]
+
+  labels {
+    env                         = "foo"
+    default_table_expiration_ms = 3600000
+  }
+}`, datasetID)
+}
+
+func testAccBigQueryDatasetWithTwoAccess(datasetID string) string {
+	return fmt.Sprintf(`
+resource "google_bigquery_dataset" "access_test" {
+  dataset_id                  = "%s"
+  friendly_name               = "foo"
+  description                 = "This is a foo description"
+  location                    = "EU"
+  default_table_expiration_ms = 3600000
+
+  access = [
+    {
+	role          = "OWNER"
+	user_by_email = "Joe@example.com"
+    },
+    {
+	role	      = "READER"
+	domain	      = "example.com"
+    }
+  ]
+
+  labels {
+    env                         = "foo"
+    default_table_expiration_ms = 3600000
+  }
+}`, datasetID)
+}
+
+func getBigQueryTableWithView(datasetID, tableID string) string {
+	return fmt.Sprintf(`
+resource "google_bigquery_dataset" "other_dataset" {
+  dataset_id = "%s"
+}
+
+resource "google_bigquery_table" "table_with_view" {
+  table_id   = "%s"
+  dataset_id = "${google_bigquery_dataset.other_dataset.dataset_id}"
+
+  time_partitioning {
+    type = "DAY"
+  }
+
+  view {
+        query = "SELECT state FROM [lookerdata:cdc.project_tycho_reports]"
+        use_legacy_sql = true
+  }
+}`, datasetID, tableID)
+}
+
+func testAccBigQueryDatasetWithViewAccess(datasetID, otherDatasetID, otherTableID string) string {
+	projectID := getTestProjectFromEnv()
+	otherTable := getBigQueryTableWithView(otherDatasetID, otherTableID)
+	// Note that we have to add a non-view access to prevent BQ from creating 4 default
+	// access entries.
+	return fmt.Sprintf(`
+%s
+
+resource "google_bigquery_dataset" "access_test" {
+  dataset_id                  = "%s"
+  friendly_name               = "foo"
+  description                 = "This is a foo description"
+  location                    = "EU"
+  default_table_expiration_ms = 3600000
+
+  access = [
+    {
+	role          = "OWNER"
+	user_by_email = "Joe@example.com"
+    },
+    {
+	view = {
+		project_id = "%s"
```

Comment:
could this be `project_id = "${google_bigquery_dataset.other_dataset.project}"`? would that make sense semantically?